pkg/v3/service/recoverer.go

```diff
@@ -0,0 +1,86 @@
+package service
+
+import (
+	"context"
+	"log"
+	"runtime/debug"
+	"time"
+)
+
+const (
+	panicRestartWait = 10 * time.Second
```

Comment:
can we discuss if we really want to recover from panics instead of failing hard? Is this a pattern used in other places in core node?